public/js/addTask.js

```diff
@@ -7,24 +7,25 @@ async function taskFormHandler(event) {
       window.location.toString().split('/').length - 1
     ];
   
-    if (task_name) {
+    if (name && volunteer && event_id) {
         const response = await fetch('/api/tasks', {
           method: 'POST',
           body: JSON.stringify({
-            event_id,
-            name
+            name,
+            volunteer,
```

Comment:
The volunteer information won't be available until a person selects this task to volunteer for.